### PR TITLE
92/Splunk support functions

### PR DESCRIPTION
**Pull request by MichaelMcAleer**
This pull request makes a number in changes to introduce new functions for Splunk support.  All unit tests are running clean including pep8 and pylint. Impacted CI tests are running cleanly, full CI run will be run overnight.

- Masking View performance support has been added to `performance.py`
    - `PyU4V.performance.PerformanceFunctions.get_masking_view_keys`
    - `PyU4V.performance.PerformanceFunctions.get_masking_view_stats`
- Director list, Port list, IP interface list, and get IP interface details have been added to `system.py`
    - `PyU4V.system.SystemFunctions.get_director_list`
    - `PyU4V.system.SystemFunctions.get_director_port_list`
    - `PyU4V.system.SystemFunctions.get_ip_interface_list`
    - `PyU4V.system.SystemFunctions.get_ip_interface`
- Array storage level details are now available via `provisioning.py`
    - `PyU4V.provisioning.ProvisioningFunctions.get_array`
- Array level WLP capabilities are now accessible in `workload_planner.py`
    - `PyU4V.workload_planner.WLPFunctions.get_capabilities`
- `array_id` is now an optional parameter for the following functions:
    - `PyU4V.replication.ReplicationFunctions.get_array_replication_capabilities`
    - `PyU4V.migration.MigrationFunctions.get_migration_info`
    - `PyU4V.migration.MigrationFunctions.get_array_migration_capabilities`

### CODE REVIEW

File: PyU4V/tests/ci_tests/test_pyu4v_ci_system.py [hunk 1]
```diff
@@ -502,3 +502,93 @@ def test_download_audit_log_record_return_binary(self):
         self.assertTrue(response.get(SUCCESS))
         self.assertIn(BINARY_DATA, response.keys())
         self.assertIsInstance(response.get(BINARY_DATA), bytes)
+
+    def test_get_director_list(self):
+        """Test get_iscsi_director_list."""
```
Comment (helenwalsh) [inline PyU4V/tests/ci_tests/test_pyu4v_ci_system.py]: nit 
""" Test get_director_list for iscsi."""  or """Test get_director_list."""
whichever is appropriate

File: PyU4V/tests/ci_tests/test_pyu4v_ci_system.py [hunk 2]
```diff
@@ -502,3 +502,93 @@ def test_download_audit_log_record_return_binary(self):
...
+        response = self.system.get_director_list()
+        self.assertTrue(response)
+        self.assertIsInstance(response, list)
+
+    def test_get_director_list_iscsi_only(self):
+        """Test get_iscsi_director_list."""
```
Comment (helenwalsh) [inline PyU4V/tests/ci_tests/test_pyu4v_ci_system.py]: See comment above.

File: PyU4V/tests/ci_tests/test_pyu4v_ci_system.py [hunk 3]
```diff
@@ -502,3 +502,93 @@ def test_download_audit_log_record_return_binary(self):
...
+    def test_get_director_port_list(self):
+        """Test get_director_port_list."""
+        director_list = self.system.get_director_list()
+        director_id = random.choice(director_list)
+        response = self.system.get_director_port_list(director_id=director_id)
+        self.assertTrue(response)
+        self.assertIsInstance(response, list)
+
+    def test_get_director_port_list_iscsi_target_set(self):
+        """Test get_director_port_list."""
```
Comment (helenwalsh) [inline PyU4V/tests/ci_tests/test_pyu4v_ci_system.py]: """Test get_director_port_list iSCSI target set."""  or something like this

File: PyU4V/tests/unit_tests/test_pyu4v_performance.py [hunk 4]
```diff
@@ -1275,6 +1275,21 @@ def test_get_iscsi_target_stats(self):
         self.assertEqual(response.get('reporting_level'),
                          self.common.convert_to_snake_case(pc.ISCSI_TGT))
 
+    def test_get_masking_view_keys(self):
+        """Test get_port_group_keys."""
```
Comment (helenwalsh) [inline PyU4V/tests/unit_tests/test_pyu4v_performance.py]: """Test get_masking_view_keys."""

File: PyU4V/tests/unit_tests/test_pyu4v_performance.py [hunk 5]
```diff
@@ -1275,6 +1275,21 @@ def test_get_iscsi_target_stats(self):
...
+        response = self.perf.get_masking_view_keys()
+        self.assertIsInstance(response, list)
+        self.assertTrue(response[0].get(pc.MV_ID))
+
+    def test_get_masking_view_stats(self):
+        """Test get_port_group_stats."""
```
Comment (helenwalsh) [inline PyU4V/tests/unit_tests/test_pyu4v_performance.py]: """Test get_masking_view_stats."""

File: PyU4V/tests/unit_tests/test_pyu4v_system.py [hunk 6]
```diff
@@ -476,3 +476,55 @@ def test_download_audit_log_record_write_file_no_name(self, mck_write):
             mck_write.assert_called_once()
             self.assertTrue(response[SUCCESS])
             self.assertIn('/test/test.pdf', str(response[AUDIT_RECORD_PATH]))
+
+    def test_get_director_list(self):
+        """Test get_iscsi_director_list."""
```
Comment (helenwalsh) [inline PyU4V/tests/unit_tests/test_pyu4v_system.py]: """Test get_director_list."""

File: PyU4V/tests/unit_tests/test_pyu4v_system.py [hunk 7]
```diff
@@ -476,3 +476,55 @@ def test_download_audit_log_record_write_file_no_name(self, mck_write):
...
+        array_id = self.data.array
+        dir_list = self.system.get_director_list(array_id=array_id)
+        self.assertTrue(dir_list)
+        self.assertIsInstance(dir_list, list)
+        self.assertEqual([self.data.director_id1, self.data.director_id2],
+                         dir_list)
+
+    def test_get_iscsi_director_list(self):
+        """Test get_iscsi_director_list."""
```
Comment (helenwalsh) [inline PyU4V/tests/unit_tests/test_pyu4v_system.py]: """Test get_director_list iSCSI."""

**Review by helenwalsh — CHANGES_REQUESTED** [review]
Few minor docstrings in the test files.

**Comment by MichaelMcAleer** [comment]
Thanks Helen, all those comments have been addressed in the most recent commit.

**Review by rawstorage — APPROVED** [review]

**Review by helenwalsh — APPROVED** [review]
Thanks Michael,
CI and unit tests clean, I approve

**Pull request merged**